src/engine/SCons/Action.py

```diff
@@ -477,6 +477,49 @@ def execute(self, target, source, env):
 
         cmd_list, ignore, silent = self.process(target, map(rfile, source), env)
 
+        # DLADD: JRH 20151022 SF37871/SF37872: DLE now contains to many .java and .cs files
+        #                                      to compile with the standard windows cmd.exe
+        # on windows, use powershell when available/necessary (to avoid command line length limit failures)
+        if (sys.platform == 'win32') and (cmd_line[0] == 'javac'):
+            try:
+                maxline = int(env.subst('$MAXLINELENGTH'))
+            except ValueError:
+                maxline = 2048
+
+            psc_key = None
+            try:
+                if sys.version_info < (3,0):
```

Comment:
The standard way to do this in python is to put the import in a try block, and catch ImportError when it is raised, then import the fallback version.
But, this SCons can't be run in anything newer than 2.5, so it's superfluous here, and SCons 2.4.x doesn't support python 3.